### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER]: On page load, focus MUST be set on the h1

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#11907]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Upon select Profile and arriving on the Your Profile screen, focus is automatically set on the `h2` Personal and contact information. This is likely to be confusing for non-sighted users who just selected Profile. The pattern on VA.gov is to set focus on the `h1` based on user action, so here the focus **should** be on Your profile. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to select the Profile page in the menu and know that I have arrived on that page.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov in browser
2. Start screen reader
3. Log in as user 1
4. Select Profile from the upper menu
5. Verify that the screen reader focus begins on Personal and contact information

## Possible Fixes (optional)

**Current code**
```diff
  <h1 class="vads-u-font-size--h4">Your profile</h1>
  <!-- etcetera -->
- <h2 tabindex="-1" class="vads-u-margin-y--2 medium-screen:vads-u-margin-bottom--4 medium-screen:vads-u-margin-top--3" data-focus-target="true">Personal and contact information</h2>
```

**Potential code**
```diff
+ <h1 tabindex="-1" class="vads-u-font-size--h4">Your profile</h1>
  <!-- etcetera -->
  <h2 class="vads-u-margin-y--2 medium-screen:vads-u-margin-bottom--4 medium-screen:vads-u-margin-top--3" data-focus-target="true">Personal and contact information</h2>
```

**Comment by mshea0606** [comment]
@tressaellen and @erikphansen to meet with Jennifer

**Comment by erikphansen** [comment]
![image](https://user-images.githubusercontent.com/20728956/89673866-ff1f6700-d89b-11ea-9f8f-0149dd690bfb.png)

@jenstrickland Right now as you navigate through the pages of the Profile, the focus is set on the H2 when the section loads, which seems like the more relevant headline. If we always set the focus to the H1, as you move from one section of the Profile to another screen readers would just keep saying "Your profile" instead of "Personal and contact information" and then "Direct deposit information".

**Comment by jenstrickland** [comment]
I think setting focus on the h2 after a user selects an item in the sidenav makes sense. On initial load, after the person selected Profile from the top menu, I think focus should be on the h1 (including on mobile).

**Comment by erikphansen** [comment]
So it should follow that if someone goes straight to a Profile page by entering the URL `https://staging.va.gov/profile/military-information`, for example, the H1 Your Profile should be focused. But if they got to that page via the sub nav we focus on the H2?

**Comment by erikphansen** [comment]
...or maybe not.

> As a screen reader user, I want to select the Profile page in the menu and know that I have arrived on that page.

If that is the logic, it's really about _how_ you got the page you are now at. So if you come straight to the Direct Deposit section from a link on a static page, we want to focus on the H2. Gonna have to think about how this should really work...

**Comment by jenstrickland** [comment]
> If that is the logic, it's really about _how_ you got the page you are now at. So if you come straight to the Direct Deposit section from a link on a static page, we want to focus on the H2. Gonna have to think about how this should really work...

If it is possible to set focus based on how the person arrived on the page, that would be great! I think starting with Your Profile doesn't hurt in an MVP, because it doesn't take much to tab to the additional context.

**Comment by erikphansen** [comment]
Notes for myself or @sanlouise : I think that the only change we'd need to make is to set the focus on the H1 _if the user got to the Profile via the root `/profile` route_. In other words, if this redirect is used, we need to set the focus on the H1 and not the H2: https://github.com/department-of-veterans-affairs/vets-website/blob/4b0e99baada9dfa7f2b50c1420a35a39384e4d08/src/applications/personalization/profile-2/components/Profile2Router.jsx#L174-L178

This way:
- users who get to the Profile by going to va.gov/profile (either directly or via the main site nav) will have the focus set on the H1 as Jen requested in this ticket
- users who get to any Profile page via a "deep link" like va.gov/profile/direct-deposit or even va.gov/profile/personal-information will have the focus set on that section's title, the H2, like it behaves now.

**Comment by jenstrickland** [comment]
I'd feel comfortable if we have @1Copenut review this. I want to be sure that a non-sighted user knows where they are, and that we're not missing something.

**Comment by 1Copenut** [comment]
@jenstrickland & @erikphansen I'm happy to take a look. I'm reading through the comment thread at EOD Friday, and will put this on a screen reader Monday as part of my morning.

**Comment by erikphansen** [comment]
This should be resolved now. Focus is on the "Your profile" H1 if you get to the Profile via the main nav.

**Comment by erikphansen** [comment]
@jenstrickland should validate that this works correctly.

**Comment by jenstrickland** [comment]
Validated. Thank you!
